### ISSUE TITLE
relative excludes 

### ISSUE THREAD

**Issue by JoyKrishnaMondal** [substack/node-browserify#1565]
```
- src
  - ./header.js
  - main.js <- browserify target this file - has requires ('./header.js')
```

now in header

```javascript

# ./header.js

var foo_bar = require ('foo-bar')

```
so far so good.
now for browserify I explicitly ask browserify to exclude `./header.js`

so lets look at foo-bar
```
foo-bar directory

- node_modules
   - foo-bar
     - dist
       - header.js
       - main.js <- entry file has require ('./header.js')
```

browserify unfortunately also ignores `./header.js` in foo bar :(

@substack - what do ?

cheers !

**Comment by MellowMelon** [comment]
Browserify's exclude attempts to deal with both paths (what you want) and the actual arguments to require (what you're doing now). Instead of excluding `./header.js`, try excluding the path to the file.

Excluding/externalizing by require argument is useful in cases where you want to prevent requiring npm modules, e.g. to create a separate bundle with third party code that won't be changing.

**Comment by JoyKrishnaMondal** [comment]
@MellowMelon  

yes that is exactly what I am trying to achieve ! - multiple bundles

if you do 

exclude:'C:/working/src/header.js'

you also need to in your source code do something like this
```
require (''C:/working/src/header.js'')
```

something I want to avoid.

**Comment by MellowMelon** [comment]
Don't exclude the absolute path; exclude the relative path from the `basedir` (which defaults to the directory from which you ran browserify). You do not need to change the require arguments.

**Comment by JoyKrishnaMondal** [comment]
that is exactly what I was doing :( unfortunately it seems browserify does a string match for excludes.

**Comment by MellowMelon** [comment]
It does both. Check your path again.
